### ISSUE TITLE
Error of build when a field ```models``` is used in a schema

### ISSUE THREAD

**Issue by LeoAutomateSAS** [francescov1/mongoose-tsgen#128]
Hello,

it appears that when a field ```models``` is used as a standard schema fields, tsgen interpretes it in a wrong way (as a reserved word ?) : 

This definition fails : 
```js
const LandingPageSchema = new Schema({
    brand: String,
    models: [
        {
            name: String,
            imgUrl: String,
            modelUrl: String,
            metadata: {
                title: String,
                description: String,
                keywords: [String],
            },
        },
    ],
})
```

with ts-gen outputs : 
![image](https://github.com/francescov1/mongoose-tsgen/assets/138013668/55810897-7898-4337-94ea-d7d095483c8d)
![image](https://github.com/francescov1/mongoose-tsgen/assets/138013668/7ff1dcc7-ea9f-4330-a593-4188ebf8d733)

This definition works (after changing the ```models``` field): 
```js
const LandingPageSchema = new Schema({
    brand: String,
    brandsModels: [
        {
            name: String,
            imgUrl: String,
            modelUrl: String,
            metadata: {
                title: String,
                description: String,
                keywords: [String],
            },
        },
    ],
})
```

tsgen installed version : 9.2.7

**Comment by francescov1** [comment]
Hey @LeoAutomateSAS, yes the conflict is because we name our models `{collection name}Model`. Let me look into this and see if there's an easy fix to check for conflicts and mitigate them.

**Comment by francescov1** [comment]
Hi @LeoAutomateSAS, I found a solution. Essentially I can check if a field conflicts with the model name and append a string to it to avoid the conflict.

So in effect, your type would result in something like this:

```typescript
export type LandingPage = {
  brand?: string;
  models: LandingPageModel{post fix to add}[];
  _id: mongoose.Types.ObjectId;
};
```

Any thoughts on what youd want the post fix to be? I havent had this use case from anyone else so would love some input. I was thinking `Subdoc`, but its a bit weird because we would end up generating a `LandingPageModelSubdocDocument` type. Maybe `Field`? So it would be named `LandingPageModelField` and `LandingPageModelFieldDocument`.

**Comment by francescov1** [comment]
Fix is live in v9.2.9!
